### ISSUE TITLE
Ember Cli is not installed

### ISSUE THREAD

**Issue by g5codyswartz** [felixrieseberg/vsc-ember-cli#37]
When I open up VSCode on mac I keep getting `Ember Cli is not installed`
When I type into the terminal `ember -v` it returns: 
```
ember-cli: 2.9.1
node: 6.9.1
os: darwin x64
```

I'm using nvm, I'm not sure if that would matter or not?

**Comment by g5codyswartz** [comment]
I ran: 
```
> nvm deactivate
> ember -v 
Node Sass could not find a binding for your current environment: OS X 64-bit with Node.js 7.x
> npm rebuild node-sass
```
now `ember -v` is displaying information
```
WARNING: Node v7.2.0 has currently not been tested against Ember CLI and may result in unexpected behaviour.
ember-cli: 2.9.1
node: 7.2.0
os: darwin x64
```
I'm still getting the error though `Ember Cli is not installed` when I restart VSCode. I'm not sure what scope/context it's running in.

**Comment by g5codyswartz** [comment]
I've also tried to add the environment variable but it doesn't seem to be showing up in the output when I start up VCS

**Comment by chazkii** [comment]
@g5codyswartz I also ran into the same problem, so looked for the particular ember binary it uses which is the one located in local `node_modules`, if it exists. https://github.com/felixrieseberg/vsc-ember-cli/blob/master/src/config.ts#L64. I ran it and got a missing package error. Did an `npm install` and now vsc-ember-cli is working again.

**Comment by g5codyswartz** [comment]
@chuckus I only have nvm installed and no system-wide node installed so that I do not run into weird conflicts (I used to have system, brew, and nvm installed which made things pretty crazy)

I noticed the error is now gone. I'm not sure if the extension changed or something on my system. I recently fixed up `rvm` using this method that I created a blog article about [RVM is not a function in VSCode](http://codyswartz.us/wp/mac/rvm-is-not-a-function-in-vscode)

TLDR; add this to your settings for mac to use a login shell
```
"terminal.integrated.shellArgs.osx": [
  "-l"
]
```

I imagine that `rvm` and `nvm` function similar. I tried removing my user config and still did not receive that error though ¯\\\_(ツ)_/¯

I always figured the extension was looking in some hard-coded area rather than a dynamic path, or maybe it was similar to using the login shell vs not?

Whatever it be, it seems to be fixed now.

**Comment by t-sauer** [comment]
I am pretty sure that this problem still exists.  
I also get this error message from time to time (the last time was earlier today) but I am not able to reproduce it consistently.

**Comment by g5codyswartz** [comment]
I just experienced it when opening up my full on ember app (other app had ember parts included and wasn't just only an ember app)

**Comment by g5codyswartz** [comment]
Seems I forgot to post this last week (just found this tab)

Seems that the version check gets around the debugger check too which is why I haven't been able to debug this issue when I try to set the debug var https://github.com/felixrieseberg/vsc-ember-cli/blob/cdc9ca806422c009c33d940367c45857afc02599/src/ember-ops.ts

Is there a way to use a login shell with the code above?

**Comment by ctsstc** [comment]
I'm also getting this on my Windows 10 machine. Running VSCode as administrator. Already ran `npm install -g ember-cli`

**Comment by ctsstc** [comment]
May be fixed by https://github.com/felixrieseberg/vsc-ember-cli/pull/43

**Comment by mirkonz** [comment]
I'm still having this issue. 
It looks like a timing issue. If I switch between projects it starts working.
FYI: I'm using nvm.

**Comment by finferflu** [comment]
I'm still having this issue, and I'm using nvm as well.

**Comment by dpatz** [comment]
Would it be possible to just add a setting to point to the path?

**Comment by nathanpalmer** [comment]
Same issue here: `macOS 10.12.6`

I use `nvm` but there is also a system-wide node installed and both of them have ember-cli installed globally. If I just start up vscode it tells me it can't find ember-cli. If I start it up from the command line like this then it works.

```
cd ~/source-directory
nvm use
code .
```

Though this is rarely how I open it up.

**Comment by LeviDooley** [comment]
I'm also having this issue, and I do not use nvm. I've installed ember-cli through git-bash though. That might be related.

**Comment by pybt** [comment]
Remove `<project-folder>/node_modules/.bin/ember` fix this issue on Mac OS

**Comment by mirkonz** [comment]
@pybt Doesn't seem to work for me. Still getting the same 'Ember Cli is not installed' message.

**Comment by pybt** [comment]
@mayko780 After you deleted did you restart vscode you have to installed ember-cli global.

**Comment by monovertex** [comment]
I'm encountering this issue on a Linux Mint machine, with Ember installed globally via yarn. In the VSCode terminal, `ember -v` has the following output:
```
ember-cli: 2.14.0
node: 7.10.1
os: linux x64
```

**Comment by HelloMihai** [comment]
also running into this issue every time I'm switching project folders
```
osx: 10.13.2 (17C88)
vscode: Version 1.19.1

ember -v:
ember-cli: 2.18.0
node: 6.11.0
os: darwin x64

which -a ember
/usr/local/bin/ember
```

i noticed https://github.com/felixrieseberg/vsc-ember-cli/blob/d6d8b8395d83f2e6a89414b21e2bc8361a582007/src/config.ts#L64 is looking for `node_modules/.bin/ember` but the `.bin` folder isnt there
`ls: .vscode/extensions/felixrieseberg.vsc-ember-cli-0.3.3/node_modules/.bin: No such file or directory`

**Comment by dadreggors** [comment]
I am having the same issue on Linux (Fedora 27) with ember-cli installed globally.

$ ember -v
ember-cli: 2.18.1
node: 8.9.4
os: linux x64

$ which -a ember
/usr/bin/ember

**Comment by stormwarning** [comment]
Does #46 fix this issue?

**Comment by ctsstc** [comment]
Not sure what's holding up that merge, or if this project has been abandoned.

**Comment by lolmaus** [comment]
Experiencing the same on Windows.

**Comment by isAlmogK** [comment]
I'm also getting the issue, it's only happing in Visual Studio code. It's fine if I use my terminal

**Comment by isAlmogK** [comment]
Has anyone been able to fix this?

**Comment by driesdl** [comment]
I'm experiencing this on Windows 10 still. Using nvm as well

**Comment by tylerbecks** [comment]
I'm also having this problem

**Comment by BabyMaybe** [comment]
Also experiencing this on Linux

**Comment by ncoquelet** [comment]
Same here if I have many folders in my workspace.
With only one folder, no issue.
